pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -181,6 +180,27 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, nil
 	}
 
+	// Get the private key that will be used to configure the instance
+	privateKey, err := secrets.GetPrivateKey(kubeTypes.NamespacedName{Namespace: r.opNamespace,
+		Name: secrets.PrivateKeySecret}, r.client)
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Private key was removed, requeue
+			return reconcile.Result{}, errors.Wrapf(err, "%s does not exist, please create it", secrets.PrivateKeySecret)
+		}
+		return reconcile.Result{}, errors.Wrapf(err, "unable to get secret %s", request.NamespacedName)
+	}
+	// Update the signer with the existing privateKey
+	newSigner, err := signer.Create(privateKey)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "error creating signer")
+	}
+	r.signer = newSigner
```

Comment:
we only really need to update this if `newSigner != r.signer` - maybe it's helpful to log when this update is happening?